Add VB FileCleanupTestsBase plus dependent tests REVIEW FIRST

paul1956 opened the pull request:
Fixes #11031 by adding more tests

## Proposed changes
Add more tests to VB Runtime
This also adds a new test class to provide cleaner file cleanup, that is different than what C# provides in FileCleanupTestBase, this class will get additional functionality in another PR.

## Customer Impact
Improve code quality

## Regression? 
- No

## Risk
None just adds tests

## Test methodology
- Visual Studio Tests

## Test environment(s)
- Visual Studio

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11987)

LeafShi1 commented on src/System.Drawing.Common/src/Windows/Win32/PInvoke.SHGetStockIconInfo.cs:
```diff
@@ -20,7 +20,7 @@ internal partial class PInvoke
     ///   <para>
     ///    <see href="https://learn.microsoft.com/windows/win32/api/shellapi/nf-shellapi-shgetstockiconinfo#parameters">
-    ///     Read more on learn.microsoft.com.
+    ///     Read more on http://learn.microsoft.com.
```
Although this will take us to the website https://learn.microsoft.com, the correct URL should be `https`.

paul1956 replied:
@LeafShi1 
Fixed ones I caused, there are a lot of http: URLs in the Repo, ones I didn't create I left alone. Someone might want to review them.

LeafShi1 reviewed (commented):
LGTM with one comment

paul1956 commented:
> LGTM with one comment

@LeafShi1
Http: changed to Https: in all the places I changed. It still exists in other places I am not comfortable changing.

Tanya-Solyanik commented on src/System.Drawing.Common/src/System/Drawing/Imaging/Effects/ColorMatrixEffect.cs:
```diff
@@ -10,8 +10,10 @@ namespace System.Drawing.Imaging.Effects;
 ///  <para>
-///   See <see href="https://learn.microsoft.com/windows/win32/gdiplus/-gdiplus-recoloring-use">Recoloring</see> for
-///   examples of using a color matrix to adjust the colors of an image.
+///   <see href="https://learn.microsoft.com/windows/win32/gdiplus/-gdiplus-recoloring-use">
```
Please factor out this change, I'm not sure if it improves readability

paul1956 replied:
Removed change

Tanya-Solyanik reviewed (commented):
It will be easier to move this PR along if you factor out `Microsoft.VisualBasic.Logging` changes into a stand alone PR and rebase this branch on the comment/spelling/line length changes later.

paul1956 commented:
> It will be easier to move this PR along if you factor out `Microsoft.VisualBasic.Logging` changes into a stand alone PR and rebase this branch on the comment/spelling/line length changes later.

@Tanya-Solyanik I have no idea how to do above, my success with rebase has been 0. The logging changes don't change logic they just move stuff to new files based on @LeafShi1 PR feedback.

I would prefer to just wait till #12105 is merged to review this, but I am open to learning better methods.

Tanya-Solyanik commented:
>The logging changes don't change logic they just move stuff to new files based on @LeafShi1 PR feedback.

Sure, this is a good feedback and the right thing to do, but it's not part of test improvements that are the focus of this PR. It's better to keep PRs focused on a single change. Just keep my feedback in mind for the next PR. I'll review this one as is.

Tanya-Solyanik commented on src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/FileIoProxyTests.vb:
```diff
@@ -0,0 +1,712 @@
+        Public Sub FileExistsProxyTest()
+            Dim sourceDirectoryName As String = CreateTempDirectory()
+            Dim sourceFileName As String = CreateTempFile(sourceDirectoryName, NameOf(sourceFileName), size:=1)
+            _fileSystem.FileExists(sourceFileName).Should.BeTrue()
```
Compare against `FileIO.FileSystem` result because we want to establish that they are doing the same thing.

Tanya-Solyanik commented on src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/FileIoProxyTests.vb:
```diff
@@ -0,0 +1,712 @@
+
+        <WinFormsFact>
+        Public Sub DirectoryExistsProxyTest()
+            _fileSystem.DirectoryExists(BaseTempPath).Should.BeTrue()
```
Compare against FileIO.FileSystem result because we want to establish that they are doing the same thing.

paul1956 replied:
@Tanya-Solyanik in this case I need to ensure BastTempPath exists, but I can add another test to compare that the proxy also matches

Tanya-Solyanik commented on src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/FileIoProxyTests.vb:
```diff
@@ -0,0 +1,712 @@
+        End Sub
+
+        <WinFormsFact>
+        Public Sub FileNormalizePathEmptyStringTest_Fail()
```
We don't own this exception, it should be tested by the System.IO.

paul1956 replied:
@Tanya-Solyanik Are you saying its sufficient to test the throw and not test the message?

Tanya-Solyanik commented on src/Microsoft.VisualBasic.Forms/tests/UnitTests/System/Windows/Forms/FileIoProxyTests.vb:
```diff
@@ -0,0 +1,712 @@
+
+Namespace Microsoft.VisualBasic.Forms.Tests
+
+    Public Class FileIoProxyTests
```
This file is testing System.IO functionality, that we don't own. I feel that some tests can be removed.

paul1956 replied:
@Tanya-Solyanik I am testing FileSystemProxy which is a very thin layer over FileIO.FileSystem which is a thin layer over System.IO. The majority of FileSystemProxy functions only have 1 test, a few have more but they either have conditional branches or they are uses are part of another test that assume they work, or they are also tested in C#.

The ones that overlap with existing tests in FileSystemProxyTests in C#, I can remove, or they could be removed from C# as those are not Fluent. The C# code does not yield 100% coverage.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/SingleInstanceHelpers.vb:
```diff
@@ -21,7 +21,9 @@ Namespace Microsoft.VisualBasic.ApplicationServices
                     If bytesRead = 0 Then
                         Exit While
                     End If
+#Disable Warning CA1849 ' Call async methods when in an async method
```
Its fixed in a later PR but I will bring it here.

paul1956 commented on src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Logging/FileLogTraceListener.ReferencedStream.vb:
```diff
@@ -50,17 +50,21 @@ Namespace Microsoft.VisualBasic.Logging
+            '''  Indicates whether or not the <see cref="Stream"/> is still in use by a FileLogTraceListener.
             ''' </summary>
-            ''' <value><see langword="True"/> if the stream is being used, otherwise <see langword="False"/>.</value>
+            ''' <value>
```
Yes that was an error.

paul1956 commented:
@Tanya-Solyanik I also realized the name of the file and class for FileIoProxyTests is wrong, it should be FileSystemProxyTests. Should I rename it or leave it?

ricardobossan commented on src/System.Windows.Forms/src/System/Windows/Forms/Input/InputLanguage.cs:
```diff
@@ -109,7 +109,8 @@ public string LayoutName
-    ///  Returns the <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
+    ///  Returns the
+    ///  <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
     ///  keyboard layout identifier</see> of the current input language.
```
I think this better follows the pattern of the other see tags in the source code:

```cs
    /// <summary>
    ///  Returns the
    ///  <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
    ///  keyboard layout identifier
    ///  </see>
    ///  of the current input language.
    /// </summary>
```

paul1956 replied:
@ricardobossan The should be addressed in a separate PR I will open. This one already has too many changes unrelated to the to FileCleanupTestBase. I will open another PR just to address "See href" based on Main. That could be merged before or after this one.

ricardobossan commented on src/System.Windows.Forms/src/System/Windows/Forms/Input/InputLanguage.cs:
```diff
@@ -109,7 +109,8 @@ public string LayoutName
-    ///  Returns the <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
+    ///  Returns the
+    ///  <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
     ///  keyboard layout identifier</see> of the current input language.
```
I think this better follows the pattern of the other see tags in the source code:

```cs
    /// <summary>
    ///  Returns the
    ///  <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeyboardlayoutnamew">
    ///  keyboard layout identifier
    ///  </see>
    ///  of the current input language.
    /// </summary>
```

lonitra commented on src/System.Windows.Forms.Primitives/src/System/Windows/Forms/AccessibiltyExtensions.cs:
```diff
@@ -11,7 +11,7 @@ internal static unsafe class AccessibiltyExtensions
     internal static LRESULT GetLRESULT(this IAccessible accessible, WPARAM wparam)
     {
-        // https://learn.microsoft.com/windows/win32/winauto/how-to-handle-wm-getobject
+        // https://learn.mcrosoft.com/windows/win32/winauto/how-to-handle-wm-getobject
```
```suggestion
        // https://learn.microsoft.com/windows/win32/winauto/how-to-handle-wm-getobject
```

paul1956 replied:
@LeafShi1 The function name AccessibiltyExtensions is misspelled, it should be AccessibilityExtensions. It is also no used should it be removed?

lonitra reviewed (commented):
LGTM apart from small typo

paul1956 commented:
@lonitra I will open a new issue on AccessibilityExtensions spelling.

paul1956 commented:
@lonitra anything preventing merge? Do I still need to fix spelling issue here?

lonitra reviewed (approved):
Looks good, thank you
